File: 12B/pa5/DictionaryTest.c
// DictionaryTest.c
// Nikolai Chen
// nmchen
// pa5
// DictionaryTest.c is a test file to test all of the operations in Dictionary.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "Dictionary.h"

int main(int argc, char* arv[]) {

	// tests Dictionary constructor
	Dictionary dict = newDictionary();


	// tests isEmpty() and size()
	printf("isEmpty: %s\n", (isEmpty(dict)?"true":"false"));
	printf("size: %d\n", size(dict));

	// tests insert() and lookup()

	insert(dict, "reach", "for the stars");
	insert(dict, "I'm", "not done yet!");
	insert(dict, "star", "wars");
	insert(dict, "YOLO", "CMPE16");
	insert(dict, "UCSC", "Sammy the Slug");

	printf("isEmpty: %s\n", (isEmpty(dict)?"true":"false"));
	printf("size: %d\n", size(dict));

	printf("key = star value = %s\n", lookup(dict, "star"));

	// tests printDictionary()

	printDictionary(stdout, dict);

	// tests delete()

	delete(dict, "I'm");
	delete(dict, "YOLO");

	printf("isEmpty: %s\n", (isEmpty(dict)?"true":"false"));
	printf("size: %d\n", size(dict));

	printf("key = YOLO value = %s\n", lookup(dict, "YOLO"));

	printDictionary(stdout, dict);

	// tests makeEmpty()

	makeEmpty(dict);

	printf("isEmpty: %s\n", (isEmpty(dict)?"true":"false"));
	printf("size: %d\n", size(dict));

	printDictionary(stdout, dict);

	exit(EXIT_SUCCESS);
}

File: 12M/lab7/DictionaryTest.java
public class DictionaryTest {

	public static void main(String args[]) {

		// tests Dictionary constructor

		Dictionary dict = new Dictionary();

		// tests isEmpty() and size()

		System.out.println("isEmpty: " + dict.isEmpty());
		System.out.println("size: " + dict.size());

		// tests insert()

		dict.insert("Halo", "Reach");
		dict.insert("Data", "Structures");
		dict.insert("Alpha", "Male");
		dict.insert("Spartan", "117");
		dict.insert("Master", "Chief");
		dict.insert("John", "Luck");

		System.out.println("isEmpty: " + dict.isEmpty());
		System.out.println("size: " + dict.size());

		// tests lookup()

		System.out.println(dict.lookup("Alpha"));
		System.out.println(dict.lookup("Male"));
		System.out.println(dict.lookup("Data"));

		// tests toString()

		System.out.println("\nDictionary in order:\n" + dict.toString());

		// tests delete()

		dict.delete("Spartan");
		dict.delete("Alpha");

		// tests insert() and delete() exceptions

		// dict.insert("Data", "Bases");
		// dict.delete("Alpha");

		System.out.println("isEmpty: " + dict.isEmpty());
		System.out.println("size: " + dict.size());
		System.out.println("\nDictionary in order:\n" + dict.toString());

		// tests makeEmpty()

		dict.makeEmpty();

		System.out.println("isEmpty: " + dict.isEmpty());
		System.out.println("size: " + dict.size());
		System.out.println("\nDictionary in order:\n" + dict.toString());


	}

}